### PR TITLE
[Elixir] list-comprehensions - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/elixir/exercises/concept/list-comprehensions/.docs/introduction.md [hunk 1]
```diff
@@ -1,10 +1,4 @@
-Comprehensions provide a facility for transforming _Enumerables_ easily and declaratively. There are three required parts to a comprehension:
-
-- _generators_:
-  - Values enumerated from structures that implement the _Enumerable_ protocol.
-  - Pattern matching expressions to destructure enumerated values.
-- _filters_: Boolean conditions, used to select which enumerated values to use when creating the new values.
-- _collectables_: A structure which implements the _Collectable_ protocol, used to collect the new values.
```
Comment (angelikatyborska) [inline languages/elixir/exercises/concept/list-comprehensions/.docs/introduction.md:7]: I remove this part because it's hard to follow without seeing any code samples first. I added an example of a filter later, and `into:` is not needed to solve this exercise, so we might skip it altogether.

File: languages/elixir/exercises/concept/list-comprehensions/.docs/introduction.md [hunk 2]
```diff
@@ -23,7 +24,7 @@ end
 # => [:b, :d]
 ```
 
-A _cartesian product_ can be created using multiple generators:
+A _cartesian product_ can be created using multiple generators. That means that each value generated by the first generator will be paired once with each value generated by the second generator:
```
Comment (angelikatyborska) [inline languages/elixir/exercises/concept/list-comprehensions/.docs/introduction.md:27]: "Cartesian product" is a mathematical term that many programmers won't know. I had to look it up too... if we want to use it, we need to explain immediately what it means in a simplified way.

**Review by neenjaw — APPROVED** [review]
Looks great!
